ext/http/01_http.js

```diff
@@ -314,30 +316,24 @@
 
           deferred.resolve([conn, res.readBuf]);
         }
-        const ws = resp[_ws];
-        if (ws) {
-          const wsRid = await core.opAsync(
+        if (resp[_ws]) {
+          if (resp[_ws].kind === null) {
+            throw new Error(
+              "No websocket was used from Deno.upgradeWebSocket() call",
+            );
+          }
+          const ws = resp[_ws].kind === "socket"
+            ? resp[_ws].socket
+            : resp[_ws].stream;
+          ws[_rid] = await core.opAsync(
             "op_http_upgrade_websocket",
             streamRid,
           );
-          ws[_rid] = wsRid;
-          ws[_protocol] = resp.headers.get("sec-websocket-protocol");
 
           httpConn.close();
 
-          if (ws[_readyState] === WebSocket.CLOSING) {
-            await core.opAsync("op_ws_close", { rid: wsRid });
-
-            ws[_readyState] = WebSocket.CLOSED;
-
-            const errEvent = new ErrorEvent("error");
-            ws.dispatchEvent(errEvent);
-
-            const event = new CloseEvent("close");
-            ws.dispatchEvent(event);
-
-            core.tryClose(wsRid);
-          } else {
```

Comment:
this whole deletion is due to this branch not even being possible; this was a mistake on my end when first implementing this.